FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -879,9 +889,11 @@ uintptr_t ContextImpl::CompileBlock(FEXCore::Core::CpuStateFrame* Frame, uint64_
     return (uintptr_t)CodePtr;
   }
 
-  // Insert to lookup cache
-  // Pages containing this block are added via AddBlockExecutableRange before each page gets accessed in the frontend
-  AddBlockMapping(Thread, GuestRIP, CodePtr);
+  if (!SkipCache) {
+    // Insert to lookup cache if TF is unset, otherwise the compiled block is a single instruction ephemeral block which shouldn't be cached
+    // Pages containing this block are added via AddBlockExecutableRange before each page gets accessed in the frontend
+    AddBlockMapping(Thread, GuestRIP, CodePtr);
```

Comment:
What does skipping this line imply in terms of ownership of compiled data? Will the instruction just permanently linger in the CodeBuffer unless the CodeBuffer gets freed for unrelated reasons?

That would probably be fine but as a personal reminder I'd need to do some chewing if this could cause any problems with code caching due to the 128 MiB limit for relative jumps.

Reply:
Yeah it'll linger until it's cleared for some reason, it's really not a problem though considering the frequency of single steps/inline SMC and the size of generated code. If it is an issue I think it's better to solve when we get to it